scripts/setup_e2e_docker.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+const {spawn, exec} = require('child_process');
+
+const axios = require('axios');
+
+const mmctlPath = process.env.MMCTL_PATH || 'mmctl';
+
+const ping = setInterval(async () => {
+    try {
+        const pingRequest = await axios.get('http://localhost:8065/api/v4/system/ping');
+        if (pingRequest.status === 200) {
+            const addUserRequest = await axios.post(
+                'http://localhost:8065/api/v4/users',
+                {
+                    email: 'test@test.com',
+                    username: 'admin1',
+                    password: 'Sys@dmin123',
+                    allow_marketing: false,
+                });
+            if (addUserRequest.status === 201) {
+                clearInterval(ping);
+
+                exec('echo "Sys@dmin123" > passfile', () => {
+                    const mmctlauth = spawn(mmctlPath, ['auth', 'login', 'http://localhost:8065', '--name', 'local-server', '--username', 'admin1', '--password-file', 'passfile']);
+                    mmctlauth.stdout.on('data', (data) => {
+                        console.log(`${data}`);
+                    });
+
+                    mmctlauth.stderr.on('data', (data) => {
+                        console.log(`ERROR: ${data}`);
+                    });
+
+                    mmctlauth.on('close', () => {
+                        const sampledata = spawn(mmctlPath, ['sampledata']);
+                        sampledata.stdout.on('data', (data) => {
+                            console.log(`${data}`);
+                        });
+
+                        sampledata.stderr.on('data', (data) => {
+                            console.log(`ERROR: ${data}`);
+                        });
+
+                        sampledata.on('close', () => {
+                            exec(`${mmctlPath} config set AnnouncementSettings.UserNoticesEnabled false`, (err, stdout, stderr) => {
+                                console.log(err, stdout, stderr);
+                            });
+                        });
+                    });
+                });
+            }
+        } else {
+            console.log(`ERROR: Trying to contact server, got ${pingRequest.status}`);
+        }
+    } catch {
+        console.log('waiting for server to respond...');
```

Comment:
we should have a max amount of retries, although most likely CI server will have a timeout eventually. 0/5 on this

Reply:
Yeah I think it just makes sense to rely on the CI server here. For example, it could take longer than expected for the image to come up due to service issues at CircleCI and i'd like their timeout to be the determining factor if something has gone wrong.